course/src/vector3.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Isometry library
+ * Copyright 2020 Ekumen Inc.
+ * Author: Alexis Pojomovsky, 2020
+ */
+
+#include <isometry/vector3.hpp>
+#include <stdexcept>
+
+namespace ekumen {
+namespace math {
+
+const Vector3 Vector3::kZero{Vector3(0., 0., 0.)};
+const Vector3 Vector3::kUnitX{Vector3(1., 0., 0.)};
+const Vector3 Vector3::kUnitY{Vector3(0., 1., 0.)};
+const Vector3 Vector3::kUnitZ{Vector3(0., 0., 1.)};
+
+Vector3::Vector3() {
+  x_ = 0.;
+  y_ = 0.;
+  z_ = 0.;
+}
+
+Vector3::Vector3(double _x, double _y, double _z) : x_{_x}, y_{_y}, z_{_z} {}
+
+Vector3 Vector3::operator+(const Vector3& vector) const {
+  return Vector3{x() + vector.x(), y() + vector.y(), z() + vector.z()};
+}
+
+Vector3 Vector3::operator-(const Vector3& vector) const {
+  return Vector3{x() - vector.x(), y() - vector.y(), z() - vector.z()};
+}
+
+Vector3 Vector3::operator*(const Vector3& vector) const {
+  return Vector3{x() * vector.x(), y() * vector.y(), z() * vector.z()};
+}
+
+Vector3 Vector3::operator*(double scalar) const {
+  return Vector3{x() * scalar, y() * scalar, z() * scalar};
+}
+
+Vector3 Vector3::operator*(int scalar) const {
+  auto scalar2 = static_cast<double>(scalar);
+  return Vector3{x() * scalar2, y() * scalar2, z() * scalar2};
+}
+
+Vector3 operator*(int scalar, const Vector3& vector) { return vector * scalar; }
+
+Vector3 Vector3::operator/(const Vector3& vector) const {
+  return Vector3{x() / vector.x(), y() / vector.y(), z() / vector.z()};
+}
+
+Vector3 Vector3::operator/(double scalar) const {
+  return Vector3{x() / scalar, y() / scalar, z() / scalar};
+}
+
+Vector3& Vector3::operator+=(const Vector3& vector) {
+  x(x() + vector.x());
```

Comment:
You have access to internal members, you don't need to go through your public interface. Better do `x_ += vector.x()`, and same below in other operators.